presto-main/src/main/java/com/facebook/presto/metadata/Metadata.java

```diff
@@ -280,6 +281,23 @@
      */
     Optional<ResolvedIndex> resolveIndex(Session session, TableHandle tableHandle, Set<ColumnHandle> indexableColumns, Set<ColumnHandle> outputColumns, TupleDomain<ColumnHandle> tupleDomain);
 
+    /**
+     * Creates the specified role in the specified connector.
+     *
+     * @param grantor represents the principal specified by WITH ADMIN statement
+     */
+    void createRole(Session session, String role, Optional<PrestoPrincipal> grantor, String catalog);
```

Comment:
I think that at some point (maybe now) these methods (grant/revoke and role related) should be moved to separate interface. Not many connectors uses them so why should they bother.